utxorpc/watch.go

```diff
@@ -37,11 +39,107 @@ func (s *watchServiceServer) WatchTx(
 ) error {
 	predicate := req.Msg.GetPredicate() // Predicate
 	fieldMask := req.Msg.GetFieldMask()
+	intersect := req.Msg.GetIntersect() // []*BlockRef
 
-	log.Printf(
-		"Got a WatchTx request with predicate %v and fieldMask %v",
-		predicate,
-		fieldMask,
+	s.utxorpc.config.Logger.Info(
+		fmt.Sprintf(
+			"Got a WatchTx request with predicate %v and fieldMask %v and intersect %v",
+			predicate,
+			fieldMask,
+			intersect,
+		),
 	)
-	return nil
+
+	// Get our points
+	var points []ocommon.Point
+	if len(intersect) > 0 {
+		for _, blockRef := range intersect {
+			blockIdx := blockRef.GetIndex()
+			blockHash := blockRef.GetHash()
+			slot := uint64(blockIdx)
+			point := ocommon.NewPoint(slot, blockHash)
+			points = append(points, point)
+		}
+	} else {
+		point := s.utxorpc.config.LedgerState.Tip().Point
+		points = append(points, point)
+	}
+
+	// Get our starting point matching our chain
+	point, err := s.utxorpc.config.LedgerState.GetIntersectPoint(points)
+	if err != nil {
+		s.utxorpc.config.Logger.Error(
+			"failed to get points",
+			"error", err,
+		)
+		return err
+	}
+
+	// Create our chain iterator
+	chainIter, err := s.utxorpc.config.LedgerState.GetChainFromPoint(*point, false)
+	if err != nil {
+		s.utxorpc.config.Logger.Error(
+			"failed to get chain iterator",
+			"error", err,
+		)
+		return err
+	}
+
+	for {
+		// Check for available block
+		next, err := chainIter.Next(false)
```

Comment:
You probably want `true` here to make the call blocking. Otherwise this is going to enter a fairly tight loop that's constantly hitting the DB while waiting for a new block.